### PR TITLE
AVT-1152: fix and review AVT-1152,AVT-1149,AVT-1151,AVT-1156

### PR DESCRIPTION
**Pull request by tranducanh**
## Description
Review and fix : AVT-1152,AVT-1149,AVT-1151,AVT-1156
(https://perzoinc.atlassian.net/browse/AVT-1149)
(https://perzoinc.atlassian.net/browse/AVT-1151)
(https://perzoinc.atlassian.net/browse/AVT-1152)
(https://perzoinc.atlassian.net/browse/AVT-1156)

## Run
Please refer attachment for more details
[Result.zip](https://github.com/symphonyoss/SymphonyElectron/files/2366296/Result.zip)

## QA Checklist
- [] Unit-Tests
- [x] Automation-Tests

### CODE REVIEW

File: tests/spectron/minimize-on-close-icon.spectron.js [hunk 1]
```diff
@@ -84,60 +90,53 @@ let app, wActions, config;
...
-    /**
-    * Verify by deselecting Minimize on Close option once the application is launched
-    * TC-ID: 3084612
-   * Cover scenarios in AVT-938
-   */
+    //     /**
+    //     * Verify by deselecting Minimize on Close option once the application is launched
+    //     * TC-ID: 3084612
+    //    * Cover scenarios in AVT-938
+    //    */
```
Comment (KiranNiranjan) [inline tests/spectron/minimize-on-close-icon.spectron.js]: Remove line comments

Comment (tranducanh) [reply tests/spectron/minimize-on-close-icon.spectron.js]: removed

File: tests/spectron/minimize-on-close-icon.spectron.js [hunk 2]
```diff
@@ -84,60 +90,53 @@ let app, wActions, config;
...
-                await Utils.sleep(2);
+        try {
+            await wActions.bringToFront("Symphony");
+            await wActions.openMinimizeAndClose(2, 1);
+            await wActions.verifyMinimizeWindows()
+            await wActions.bringToFront("Symphony");
+            await Utils.sleep(2);
+            if (!isMac)
                 await wActions.pressCtrlW();
+            else
```
Comment (keerthi16) [inline tests/spectron/minimize-on-close-icon.spectron.js]: Please include `curly brace`

File: tests/spectron/close.spectron.js [hunk 3]
```diff
@@ -29,12 +29,11 @@ describe('Tests for Close', () => {
...
 
-    it('should close the app', () => {
-        return app.stop();
-    });
-
-    it('should check whether the app is running', () => {
-        expect(app.isRunning()).toBe(false);
+    it('should check whether the app is running', async (done) => {
+        await app.stop();
+        let isRun  = await app.isRunning()
```
Comment (keerthi16) [inline tests/spectron/close.spectron.js]: Missing `;`

File: tests/spectron/minimize-on-close-icon.spectron.js [hunk 4]
```diff
@@ -59,23 +63,25 @@ let app, wActions, config;
...
-                if (userConfig.minimizeOnClose != false) {
-                    await wActions.openMenu(["Window", "Minimize on Close"]);
-                }
-                await wActions.openMenu(["Window", "Close"]);
+
+        try {
+            userConfig = await Application.readConfig(config);
+            if (userConfig.minimizeOnClose == false) {
+                await wActions.openMinimizeAndClose(1, 1);
+                await wActions.verifyMinimizeWindows()
```
Comment (keerthi16) [inline tests/spectron/minimize-on-close-icon.spectron.js]: Missing `;`

File: tests/spectron/minimize-on-close-icon.spectron.js [hunk 5]
```diff
@@ -84,60 +90,53 @@ let app, wActions, config;
...
-                await wActions.openMenu(["Window", "Close"])
-                await wActions.verifyMinimizeWindows();
 
-                await wActions.focusWindow();
-                await wActions.bringToFront("");
-                await Utils.sleep(2);
+        try {
+            await wActions.bringToFront("Symphony");
+            await wActions.openMinimizeAndClose(2, 1);
+            await wActions.verifyMinimizeWindows()
```
Comment (keerthi16) [inline tests/spectron/minimize-on-close-icon.spectron.js]: Same here

File: tests/spectron/minimize-on-close-icon.spectron.js [hunk 6]
```diff
@@ -84,60 +90,53 @@ let app, wActions, config;
...
+            await wActions.bringToFront("Symphony");
+            await Utils.sleep(2);
+            if (!isMac)
                 await wActions.pressCtrlW();
+            else
+                await wActions.pressCtrlWOnMac()
```
Comment (keerthi16) [inline tests/spectron/minimize-on-close-icon.spectron.js]: Same here

File: tests/spectron/minimize-on-close-icon.spectron.js [hunk 7]
```diff
@@ -84,60 +90,53 @@ let app, wActions, config;
...
+            await wActions.bringToFront("Symphony");
+            let count = await app.client.getWindowCount();
+            await wActions.openMinimizeAndClose(1, 1);
             try {
-                let userConfig = await Application.readConfig(config);
-                await wActions.focusWindow();
-                await wActions.openMenu(["Window", "Minimize on Close"]);
-                if (userConfig.minimizeOnClose == false) {
-                    await wActions.openMenu(["Window", "Minimize on Close"]);
+                count = await app.client.getWindowCount()
```
Comment (keerthi16) [inline tests/spectron/minimize-on-close-icon.spectron.js]: Same here

File: tests/spectron/notificationPosition.spectron.js [hunk 8]
```diff
@@ -2,20 +2,22 @@ const Application = require('./spectronSetup');
...
 
-    beforeAll((done) => {
-        return app.startApplication().then((startedApp) => {
-            app = startedApp;
-            done();
-        }).catch((err) => {
+    beforeAll(async (done) => {
+        try {
+            let testHost = await 'file:///' + path.join(__dirname, '..', '..', 'demo/index.html')
+            app = await mainApp.startApplication({testedHost:testHost, alwaysOnTop: false })
```
Comment (keerthi16) [inline tests/spectron/notificationPosition.spectron.js]: Same here

File: tests/spectron/spectronWebActions.js [hunk 9]
```diff
@@ -419,18 +419,18 @@ class WebActions {
         const bluebird = require('bluebird');
         const API_VERSION = '2.0.0';
         const SEARCH_API_VERSION = '3.0.0';
-
+        const ELECTRON_VERSION = '3.0.0'
```
Comment (keerthi16) [inline tests/spectron/spectronWebActions.js]: Same here

Comment (tranducanh) [reply tests/spectron/spectronWebActions.js]: fixed

**Pull request merged**